pints/_transformation.py

```diff
@@ -0,0 +1,1120 @@
+        q = pints.vector(q)
+        diag = np.zeros(q.shape)
+        lo = hi = 0
+        for transform in self._transforms:
+            lo = hi
+            hi += transform.n_parameters()
+            diag[lo:hi] = np.diagonal(transform.jacobian(q[lo:hi]))
+        output = np.diag(diag)
+        return output
+
+    def log_jacobian_det(self, q):
+        """ See :meth:`Transformation.log_jacobian_det()`. """
+        q = pints.vector(q)
+        output = 0
+        lo = hi = 0
+        for transform in self._transforms:
+            lo = hi
+            hi += transform.n_parameters()
+            output += transform.log_jacobian_det(q[lo:hi])
+        return output
+
+    def log_jacobian_det_S1(self, q):
+        """ See :meth:`Transformation.log_jacobian_det_S1()`. """
+        q = pints.vector(q)
+        output = 0
+        output_S1 = np.zeros(q.shape)
+        lo = hi = 0
+        for transform in self._transforms:
+            lo = hi
+            hi += transform.n_parameters()
+            j, j_S1 = transform.log_jacobian_det_S1(q[lo:hi])
+            output += j
+            output_S1[lo:hi] = np.asarray(j_S1)
+        return output, output_S1
+
+
+class IdentityTransformation(ElementWiseTransformation):
+    """
+    Identity transformation does nothing to the input parameters, i.e. the
+    search space under this transformation is the same as the model space.
+    And its Jacobian matrix is the identity matrix.
+
+    Extends :class:`ElementWiseTransformation`.
+
+    Parameters
+    ----------
+    n_parameters
+        Number of model parameters this transformation is defined over.
+    """
+    def __init__(self, n_parameters):
+        self._n_parameters = n_parameters
+
+    def jacobian(self, q):
+        """ See :meth:`Transformation.jacobian()`. """
+        return np.eye(self._n_parameters)
+
+    def jacobian_S1(self, q):
+        """ See :meth:`Transformation.jacobian_S1()`. """
+        n = self._n_parameters
+        return self.jacobian(q), np.zeros((n, n, n))
+
+    def log_jacobian_det(self, q):
+        """ See :meth:`Transformation.log_jacobian_det()`. """
+        return 0.
+
+    def log_jacobian_det_S1(self, q):
+        """ See :meth:`Transformation.log_jacobian_det_S1()`. """
+        return self.log_jacobian_det(q), np.zeros(self._n_parameters)
+
+    def n_parameters(self):
+        """ See :meth:`Transformation.n_parameters()`. """
+        return self._n_parameters
+
+    def multiple_to_model(self, qs):
+        """ See :meth:`Transformation.multiple_to_model()`. """
+        return np.asarray(qs)
+
+    def to_model(self, q):
+        """ See :meth:`Transformation.to_model()`. """
+        return pints.vector(q)
+
+    def to_search(self, p):
+        """ See :meth:`Transformation.to_search()`. """
+        return pints.vector(p)
+
+
+class LogitTransformation(ElementWiseTransformation):
+    r"""
+    Logit (or log-odds) transformation of the model parameters:
```

Comment:
Wondering if this, and all the following, should be rewritten to have the standard 1-sentence, space, 2nd paragraph format for docstrings?
The "math" part won't show up in hints in IDEs etc. So it might look like "Log (or log-odds) transformation of the model parameters:" there? Maybe just a full stop is better, and then the next paragraph can be "The transformation is given by" or something?

Reply:
Updated.